Sources_v3/Models/ChannelId.swift

```diff
@@ -9,60 +9,40 @@ import Foundation
 /// It reflects channel's type and a unique id.
 ///
 public struct ChannelId: Hashable, CustomStringConvertible {
-    private static let any = "*"
     private static let separator: Character = ":"
     
     let rawValue: String
     
-    /// Returns `true` if the channel id matches "all" channels.
-    public var isAny: Bool { rawValue == ChannelType.unknown.rawValue + String(Self.separator) + Self.any }
-    
     /// Creates a new `ChannelId` value.
     ///
     /// - Parameters:
     ///     - type: A type of the channel the `ChannelId` represents.
     ///     - id: An id of the channel the `ChannelId` represents.
     ///
     public init(type: ChannelType, id: String) {
-        rawValue = type.rawValue + "\(Self.separator)" + (id.isEmpty ? ChannelId.any : id)
+        rawValue = type.rawValue + "\(Self.separator)" + id
     }
     
     init(cid: String) throws {
-        if cid == ChannelId.any {
-            self.init(type: .unknown, id: Self.any)
-            return
-        }
-        
-        if cid.contains(ChannelId.separator) {
-            let channelPair = cid.split(separator: ChannelId.separator)
-            let type = ChannelType(rawValue: String(channelPair[0]))
-            let id = String(channelPair[1])
-            self.init(type: type, id: id)
-        } else {
+        guard !cid.isEmpty, cid.split(separator: ChannelId.separator).count == 2 else {
             throw ClientError.InvalidChannelId("The channel id has invalid format and can't be decoded: \(cid)")
         }
+        
+        rawValue = cid
```

Comment:
Why don't we store it as separated, so we don't have to `split` it every time for `type` and `id`?

Reply:
@b-onc Because then you will have to create the raw value every time you need it? I think it doesn't matter, to be honest :) It's either this or that.